### ISSUE TITLE
Unsupported type TypeReference

### ISSUE THREAD

**Issue by jasonlawless** [SolidAlloy/ClassTypeReference-for-Unity#27]
Hey there. First of all, super awesome job with this. I am very happy I found your repo. I was about to submit a much more serious issue ticket on the older repos because the dropdown wasn't working at all. Your repo seems to have fixed that issue.

Here is the issue:
If I setup a `TypeReference` field on a component, then every time ANY changes are made in the inspector to ANY field (including `TypeReference`) it throws the following:
`Unsupported type TypeReference`

The stack traces vary depending on the field that I update. But here is the stack track when I update a `TypeReference` field:
![image](https://user-images.githubusercontent.com/25869800/113221284-937a7780-9239-11eb-991d-9a201422fc22.png)

Again, the choosing of Type and using it in script during runtime works as advertised. Just keep getting these errors when changing data in editor is all.

My setup:
Unity 2020.3.1
HDRP 10.3.2
Input System 1.0.2
Odin Plugin for Inspector and Serializations

Build Settings:
Standalone. MacOS & PC x86_64
IL2CPP - Min stripping
.NET 4.7.1

If you need any other information, please let me know. I'd be happy to help you debug this issue. Thanks in advance!

**Comment by jasonlawless** [comment]
So my editor really started tweaking, it stopped persisting prefab changes I made in prefab mode, and every time I Undo anything, it would revert changes so far back and basically my Editor was on an acid trip like I've never seen.

I closed it. And opened it again, and everything worked fine, and the bug that I reported was completely gone. Everything is running smooth. It's possible something went wrong when I installed the original repo, then found a more updated repo, removed first and installed second, and then found yours, removed second and installed yours. Most likely something went wrong there and Editor cached something that wasn't there anymore.

Anywho, all is well and I apologize for opening an issue without at least trying an editor refresh. You're awesome for making this plugin! Thank you!
